src/Map/src/AssetMapper/ImportMap/Resolver/LeafletPackageResolver.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map\AssetMapper\ImportMap\Resolver;
+
+use Symfony\Component\AssetMapper\ImportMap\Resolver\PackageResolverInterface;
+use Symfony\Component\Filesystem\Path;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\UX\Map\AssetMapper\ImportMap\Compiler\LeafletReplaceImagesAssetCompiler;
+
+/**
+ * PackageResolver decorator for Leaflet.
+ *
+ * Some files mentioned in Leaflet's JavaScript code could not be detected as extra files by the actual PackageResolver.
+ * Without this decorator, in the following code, the .png files won't be detected as extra files:
+ *
+ *      // ...
+ *      $i.extend({options:{iconUrl:"marker-icon.png",iconRetinaUrl:"marker-icon-2x.png",shadowUrl:"marker-shadow.png",
+ *      // ...
+ *
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+class LeafletPackageResolver implements PackageResolverInterface
```

Comment:
Is this the only way ? Having a dedicated packageresolver just for fixing some image path feels :/

Reply:
Yeah I know, but I don't find any better solution than that. Without this custom resolver, some dependencies specific to Leaflet.js can not be resolved :/

Reply:
Yeah I know, but I don't find any better solution than that. Without this custom resolver, some dependencies specific to Leaflet.js can not be resolved :/

Reply:
Can you remind me what's the problem and when? I'll look at it as _**if**_   if _**maybe**_  i can find another way

Reply:
In Leaflet javascript code, images are mentioned like this:
```js
options: {
            iconUrl: "marker-icon.png",
            iconRetinaUrl: "marker-icon-2x.png",
            shadowUrl: "marker-shadow.png",
            iconSize: [25, 41],
            iconAnchor: [12, 41],
            popupAnchor: [1, -34],
            tooltipAnchor: [16, -28],
            shadowSize: [41, 41]
        }

There is no ESM imports, images are naively used like this:
```js
const img = document.createElement("img");
img.src = 'marker-icon.png';
```

Reply:
Also, is the class's documentation not clear at all? 🥲